src/iop/highlights.c

```diff
@@ -2138,15 +2155,23 @@ void gui_changed(dt_iop_module_t *self, GtkWidget *w, void *previous)
   dt_bauhaus_widget_set_quad_visibility(g->clip, israw);
 
   const gboolean use_laplacian = bayer && mode == DT_IOP_HIGHLIGHTS_LAPLACIAN;
-  gtk_widget_set_visible(g->noise_level, use_laplacian);
+  const gboolean use_segmentation = bayer && (mode == DT_IOP_HIGHLIGHTS_SEGMENTS);
+  gtk_widget_set_visible(g->noise_level, use_laplacian || (use_segmentation && (p->recovery > 0)));
   gtk_widget_set_visible(g->iterations, use_laplacian);
   gtk_widget_set_visible(g->scales, use_laplacian);
   gtk_widget_set_visible(g->solid_color, use_laplacian);
 
-  const gboolean use_recovery = bayer && (mode == DT_IOP_HIGHLIGHTS_SEGMENTS);
-  gtk_widget_set_visible(g->candidating, use_recovery);
-  gtk_widget_set_visible(g->combine, use_recovery);
-
+  gtk_widget_set_visible(g->candidating, use_segmentation);
+  gtk_widget_set_visible(g->combine, use_segmentation);
+  gtk_widget_set_visible(g->recovery, use_segmentation);
+  gtk_widget_set_visible(g->strength, use_segmentation && (p->recovery > 0));
+  
+  // The special case for strength button active needs further care here
+  if((use_segmentation && (p->recovery == 0)) && (g->show_segmentation == 4))
```

Comment:
`== 0` => `== DT_RECOVERY_MODE_OFF`

Also it seems that show_segmentation (despite its name we could expected a gboolean) takes different values. I see none of them documented. Maybe worth a comment in the definition in the struct?

Even better, it seems to me that this value is `dt_iop_highlights_plane_t `in `hlrecovery.c`, right? If so, please move the definition into a new `hlrecovery.h` and use this enum here (DT_IO_PLANNE_ALL).

Reply:
sure, did that

Reply:
No, it's not the plane. If you agree here i added a new enum but still in highlights.c